qa/os/src/test/java/org/elasticsearch/packaging/util/Shell.java

```diff
@@ -91,68 +113,114 @@ public Result run( String command, Object... args) {
     private Result runScript(String[] command) {
         Result result = runScriptIgnoreExitCode(command);
         if (result.isSuccess() == false) {
-            throw new RuntimeException("Command was not successful: [" + String.join(" ", command) + "] result: " + result.toString());
+            throw new RuntimeException("Command was not successful: [" + String.join(" ", command) + "]\n   result: " + result.toString());
         }
         return result;
     }
 
     private Result runScriptIgnoreExitCode(String[] command) {
         ProcessBuilder builder = new ProcessBuilder();
         builder.command(command);
-
-
         if (workingDirectory != null) {
             setWorkingDirectory(builder, workingDirectory);
         }
-
-        if (env != null && env.isEmpty() == false) {
-            for (Map.Entry<String, String> entry : env.entrySet()) {
-                builder.environment().put(entry.getKey(), entry.getValue());
-            }
+        for (Map.Entry<String, String> entry : env.entrySet()) {
+            builder.environment().put(entry.getKey(), entry.getValue());
         }
-
+        final Path stdOut;
+        final Path stdErr;
         try {
+            Path tmpDir = Paths.get(System.getProperty("java.io.tmpdir"));
+            Files.createDirectories(tmpDir);
+            stdOut = Files.createTempFile(tmpDir, getClass().getName(), ".out");
```

Comment:
Where are these files cleaned up?

Reply:
Good catch, they were not. I added cleanup.